subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -961,10 +973,14 @@ Support for these features will be added in later Gradle releases.
 [[config_cache:not_yet_implemented:secrets]]
 === Handling of credentials and secrets
 
-The configuration cache has currently no option to hide secrets that are used as inputs.
-It means that they end up in the serialized configuration cache entry.
+The configuration cache has currently no option to prevent storing secrets that are used as inputs, and so they might end up in the serialized configuration cache entry which, by default, is stored under `.gradle/configuration-cache` in your project directory.
+
+To mitigate the risk of accidental exposure, Gradle encrypts the configuration cache.
+Gradle transparently generates a machine-specific secret key as required, caches it under the
+`<<directory_layout.adoc#dir:gradle_user_home,GRADLE_USER_HOME>>` directory and uses it to encrypt the data in the project specific caches.
+
 
-This means that you should:
+To enhance security further, make sure to:
 
 * Either secure access to configuration cache entries that may contain secrets
```

Comment:
We say "either... or", but they are orthogonal, right?

Also, we say "secure access to configuration cache *entries that may contain* secrets", but users don't understand which entries may contain secrets, right? Shouldn't we advise that the entire configuration area should be secured if build configuration may contain secrets?

Reply:
Thanks, @abstratt. Feedback applied.